src/adapters/carambola.js

```diff
@@ -0,0 +1,187 @@
+
+  function _addErrorBidResponse(bid, response = {}, errorMsg = '') {
+    const bidResponse = bidfactory.createBid(2, bid);
+    bidResponse.bidderCode = BIDDER_CODE;
+    bidResponse.reason = errorMsg;
+    bidmanager.addBidResponse(_getCustomAdUnitCode(bid), bidResponse);
+  }
+  //  looking at the utils.js at getBidderRequest method. this is what is requested.
+  function _getCustomAdUnitCode(bid) {
+    return bid.placementCode;
+  }
+
+  function _addBidResponse(bid, response) {
+    const bidResponse = bidfactory.createBid(1, bid);
+    bidResponse.bidderCode = BIDDER_CODE;
+    bidResponse.ad = response.ad;
+    bidResponse.cpm = response.cpm;
+    bidResponse.width = response.width;
+    bidResponse.height = response.height;
+    bidResponse.currencyCode = response.cur;
+    bidResponse.token = response.token;
+    bidResponse.pvid = response.pageViewId;
+
+    bidmanager.addBidResponse(_getCustomAdUnitCode(bid), bidResponse);
+  }
+
+  function _getPageViewId() {
+    window.Cbola = window.Cbola || {};
+    window.Cbola.HB = window.Cbola.HB || {};
+    window.Cbola.HB.pvid = window.Cbola.HB.pvid || _createPageViewId();
+    return window.Cbola.HB.pvid;
+  }
+
+  function _createPageViewId() {
+    function _pad(number) {
+      return number > 9 ? number : '0' + number
+    }
+
+    const MIN = 10000;
+    const MAX = 90000;
+    let now = new Date();
+
+    var pvid =
+      _pad(now.getDate()) +
+      _pad(now.getMonth() + 1) +
+      _pad(now.getFullYear() % 100) +
+      _pad(now.getHours()) +
+      _pad(now.getMinutes()) +
+      _pad(now.getSeconds()) +
+      _pad(now.getMilliseconds() % 100) +
+      Math.floor((Math.random() * MAX) + MIN);
+
+    return pvid;
+  }
+
+  //  sends a request for each bid
+  function _buildRequest(bids, params) {
+    if (!utils.isArray(bids)) {
+      return;
+    }
+    //  iterate on every bid and return the  response to the hb manager
+    utils._each(bids, bid => {
+      let tempParams = params || {};
+      tempParams.cbolaMode = bid.params.cbolaMode || 0;
+      tempParams.wid = bid.params.wid || 0;
+      tempParams.pixel = bid.params.pixel || '';
+      tempParams.bidFloor = bid.params.bidFloor || 0;
+      tempParams.pageViewId = _getPageViewId();
+      tempParams.hb_token = utils.generateUUID();
+      tempParams.sizes = utils.parseSizesInput(bid.sizes) + '';
+      tempParams.bidsCount = bids.length;
+
+      for (let customParam in bid.params.customParams) {
+        if (bid.params.customParams.hasOwnProperty(customParam)) {
+          tempParams['c_' + customParam] = bid.params.customParams[customParam];
+        }
+      }
+
+      let server = bid.params.server || 'route.carambo.la';
+      let cbolaHbApiUrl = '//' + server + '/' + REQUEST_PATH;
+
+      //  the responses of the bid requests
+      ajax(cbolaHbApiUrl + _jsonToQueryString(tempParams), response => {
+        //  no response
+        if (!response || response.cpm <= 0) {
```

Comment:
`response` here is still a string and needs to be `JSON.parse`d before you can access properties e.g. `response.cpm`